### PR TITLE
chore: split agent data

### LINKED ISSUE #1: Split agent data
### Clear and concise description of the problem

Allow tree shaking agent data.

/cc @benmccann 

### Suggested solution

Split types and export also agent subpackage

### Alternative

Remove agent data.

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by userquin**
### Linked Issues
closes #1

### CODE REVIEW

**Comment by benmccann** [comment]
awesome! it could help to update the README showing how to use the new agents code

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const AGENTS = ['npm', 'yarn', 'yarn@berry', 'pnpm', 'pnpm@6', 'bun'] as const
+export type Agent = typeof AGENTS[number]
+
+// the order here matters, more specific one comes first
+export const LOCKS: Record<string, Agent> = {
```
Comment (antfu) [inline src/utils.ts:5]: Why don't we put this inside agents?

Comment (userquin) [reply src/utils.ts:5]: You're right

Comment (userquin) [reply src/utils.ts]: done

**Comment by userquin** [comment]
> awesome! it could help to update the README showing how to use the new agents code

I'll send some PR to update the docs...

**Review by antfu — APPROVED** [review]

**Pull request merged**